stl/inc/chrono

```diff
@@ -14,9 +14,14 @@
 #include <utility>
 #include <xtimec.h>
 
+#if _HAS_CXX20
+#include <compare>
+#endif // _HAS_CXX20
+
 #pragma pack(push, _CRT_PACKING)
 #pragma warning(push, _STL_WARNING_LEVEL)
 #pragma warning(disable : _STL_DISABLED_WARNINGS)
+#pragma warning(disable : 4308) // needed for constexpr date arithmetic

```

Comment:
Yeah that didn't seem right to me but didn't know how to get constexpr usage compiling.

Reply:
Can you extract a small, self-contained repro? Also, I would expect `static_cast` to be fine; I can say `static_cast<size_t>(-1)` in a `constexpr` context because that's well-defined.

Reply:
It's been a long time and I _think_ I _may_ have accidentally deleted my test code for this PR. so might have to rewrite the tests.

Speaking of which, now that they are up, would it make sense to split this paper's tests up? e.g. this PR (and potentially I/O and formatting)
`P0355R7_Calendars And Time Zones_Core`
and 
`P0355R7_Calendars And Time Zones_Clocks`
?

Reply:
This feature is large enough that multiple tests would certainly be justified, as we've done with ranges. (No spaces in directory names ever, and no capital letters outside of paper numbers, though! :smile_cat: )

Reply:
As it turns out I can't remember what caused it. So I guess I'll remove it and maybe I'll re-discover it again while I add tests.

Starting off, the tests have already spotted a bug which I've fixed, horray! Though it looks like tests 17 and 18 have trouble with `<=>`?
```
chrono(1467,65): error: invalid operands to binary expression ('const chrono::month' and 'chrono::month')
  const int _Y        = static_cast<int>(_Year) - (_Month <= chrono::month{2});
                                                   ~~~~~~ ^  ~~~~~~~~~~~~~~~~
```
With that being defined as
```cpp
constexpr strong_ordering operator<=>(const month& _Left, const month& _Right) noexcept {
    return static_cast<unsigned>(_Left) <=> static_cast<unsigned>(_Right);
}
```

Reply:
If you're encountering weird behavior with spaceship, try to reduce it to a library-free test case, so it can be reported via Developer Community to the product team. Otherwise, a self-contained test case that doesn't rely on library changes is good (either the library maintainers or the compiler devs can reduce it). If absolutely necessary, a branch of library+test changes that reproduces the suspected compiler bug can be used to extract a repro.

Reply:
Well I missed the part where it says it ignored the duration templates so probably related?
```
chrono(377,9): note: candidate template ignored: could not match 'duration<type-parameter-0-0, type-parameter-0-1>' against 'const std::chrono::month'
        operator<=(const duration<_Rep1, _Period1>& _Left, const duration<_Rep2, _Period2>& _Right) noexcept(
        ^
```

Reply:
Couldn't get a self contained test to get the same results so here's a branch https://github.com/SuperWig/STL/commit/f2a621b97371ff9c019345a9fd68d1ab72a55008

Reply:
Not entirely sure what happened but the problem with spaceship is gone now?

The static assert is now failing 17 & 18 due to
```
test.cpp(110,13): note: constexpr evaluation hit maximum step limit; possible infinite loop?
            assert(!year{i}.is_leap());
```
https://github.com/microsoft/STL/pull/323/files#diff-a4c31ccea37cd594078e07d203078552R110